.github/workflows/sketcher-builder.yml

```diff
@@ -0,0 +1,120 @@
+name: Sketcher Build and Test
+
+on:
+  workflow_dispatch: ~
+  push: ~
+  pull_request: ~
+
+env:
+  BUILD_TYPE: Release
+
+jobs:
+  # TODO: clang-format hook
+  build_test_sketcher:
+    name: ${{ matrix.config.name }}
+    runs-on: ${{ matrix.config.os }}
+    defaults:
+      run:
+        shell: bash
+    strategy:
+      fail-fast: false
+      matrix:
+        config:
+        - {
+            name: "macOS",
+            os: macos-latest,
+            app: "build/sketcher_app",
+          }
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Read external/versions.json
+        run: |
+          {
+              echo 'VERSIONS_JSON<<EOF'
+              cat external/versions.json
+              echo -e '\nEOF'
+          } >> "$GITHUB_ENV"
+
+      - name: Install Ninja
+        run: python -m pip install ninja
+
+      - name: CPU count
+        shell: python
+        run: |
+          import os
+          import multiprocessing
+          with open(os.getenv("GITHUB_ENV"), "a") as fh:
+            fh.write(f"CPU_COUNT={multiprocessing.cpu_count()}")
```

Comment:
You can remove CPU_COUNT now. `ninja` will auto detect the number of cores, and use the appropriate number of parallel build processes.

Reply:
Oh, terrific!